pkg/cimfs/cimfs.go

```diff
@@ -13,5 +15,77 @@ func IsCimFSSupported() bool {
 	if err != nil {
 		logrus.WithError(err).Warn("get build revision")
 	}
-	return osversion.Build() == 20348 && rv >= 2031
+	// TODO(ambarve): add proper version for post iron builds
+	return true || (osversion.Build() == 20348 && rv >= 2031)
+}
+
+// IsBlockCimSupported returns true if block formatted CIMs (i.e block device CIM &
+// single file CIM) are supported on the current OS build.
+func IsBlockCimSupported() bool {
+	// TODO(ambarve): add proper version here
+	return true
+}
+
+func IsMergedCimSupported() bool {
+	// TODO(ambarve): add proper version here
+	return true
+}
+
+type BlockCIMType uint32
+
+const (
+	BlockCIMTypeNone BlockCIMType = iota
+	BlockCIMTypeSingleFile
+	BlockCIMTypeDevice
```

Comment:
What's an example of when you would use one of these instead of the other?

Reply:
`BlockCIMTypeSingleFile` is used when you don't have a separate block device but still want to store the entire CIM in a block formatted way. In such cases the entire CIM is stored on a single file in the filesystem. Whereas when you want to write or use a CIM stored on something like a VHD, you would use `BlockCIMTypeDevice`. For example, you can write layers as `BlockCIMTypeSingleFile` type of CIMs on one machine. Then you can stream those layers on the container host and they will show up as block devices on the container host. You will use those streamed layers as a `BlockCIMTypeDevice` type.

Reply:
Can you give an example of a path that would be used with `BlockCIMTypeDevice`?

Reply:
Thanks. The VHD example confused me, since it can be written to both as a file, or as a mounted device.